components/script/dom/serviceworkerglobalscope.rs

```diff
@@ -215,27 +159,27 @@ impl ServiceWorkerGlobalScope {
                 return;
             }
 
-            {
-                let _ar = AutoWorkerReset::new(global.r(), worker);
-                scope.execute_script(DOMString::from(source));
-            }
+            scope.execute_script(DOMString::from(source));
 
-
-            let reporter_name = format!("service-worker-reporter-{}", random::<u64>());
-            scope.mem_profiler_chan().run_with_memory_reporting(|| {
-                // Service workers are time limited
+            // Service workers are time limited
+            spawn_named("TimeoutThread".to_owned(), move || {
```

Comment:
I made timeouts work using this, way, which seemed less involved. Am not completely sure of the api, of timers methods, that we would call here, so i did not use it. Suggestions ?